src/python/turicreate/toolkits/image_analysis/image_analysis.py

```diff
@@ -192,6 +192,171 @@ def resize(image, width, height, channels=None, decode=False, resample="nearest"
+            lambda x: _get_new_image_dimensions_with_original_aspect_ratio(x, side_length_type, side_length)
+        )
+        image_with_new_dims_sframe = _tc.SFrame({"image": image, "new_shape": new_image_shape_sarray})
+        return image_with_new_dims_sframe.apply(
+            lambda x: resize(x['image'], x['new_shape'][0], x['new_shape'][1], channels=channels, decode=decode, resample=resample)
+        )
+    else:
+        raise ValueError(
+            "Cannot call 'resize' on objects that are not either an Image or SArray of Images"
+        )
+
+
+def _get_new_image_dimensions_with_original_aspect_ratio(image, side_length_type, side_length):
+
+    image_shape = (image.width, image.height)
+    aspect_ratio = image_shape[0]/image_shape[1]
+    if aspect_ratio > 1:
+        if side_length_type == "max_length":
+            return (side_length, int(side_length/aspect_ratio))
+        else:
+            return (int(side_length*aspect_ratio), side_length)
+    else:
+        if side_length_type == "max_length":
+            return (int(side_length*aspect_ratio), side_length)
+        else:
+            return (side_length, int(side_length/aspect_ratio))
+
+
+def resize_annotations(original_image_shape, annotations, target_image_shape):
+    """
+    Resize annotations to any target image dimensions
+
+    Parameters
+    ----------
+    original_image_shape : tuple | list
+        Original image dimesions in the format of (width, height)
+    annotations : list
+        Original annotations of the data. 
+        This should be a list of dictionaries , with
+        each dictionary representing a bounding box of an object instance. Here
+        is an example of the annotations for a single image with two object
+        instances::
+
+            [{'label': 'dog',
+              'type': 'rectangle',
+              'coordinates': {'x': 223, 'y': 198,
+                              'width': 130, 'height': 230}},
+             {'label': 'cat',
+              'type': 'rectangle',
+              'coordinates': {'x': 40, 'y': 73,
+                              'width': 80, 'height': 123}}]
+
+        The value for `x` is the horizontal center of the box paired with
+        `width` and `y` is the vertical center of the box paired with `height`.
+        'None' (the default) indicates the only list column in `dataset` should
+        be used for the annotations.
+        For more information on annotations format refer to `https://apple.github.io/turicreate/docs/userguide/object_detection/` 
+    target_image_shape : tuple | list
+        Target image dimesions in the format of (width, height)
+    """
+    if type(annotations) is not list:
+        raise ValueError("annotations are expected in the form of list")
+
+    for ann_index in range(len(annotations)):
+        annotations[ann_index]["coordinates"]["height"] = annotations[ann_index]["coordinates"]["height"]*target_image_shape[1]/original_image_shape[1]
```

Comment:
Recalculating `target_image_shape[0]/original_image_shape[0]` and `target_image_shape[1]/original_image_shape[1]` twice for each element of `annotations` is not optimal. 

It's better to calculate those values once, outside of the loop, and then just use them inside the loop.

Reply:
I agree, I will fix it.